x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/OpenAiService.java

```diff
@@ -183,6 +192,13 @@ public void doInfer(
             return;
         }
 
+        if (model instanceof OpenAiChatCompletionModel) {
```

Comment:
How about we move this check to `OpenAiChatCompletionAction`. That way we can avoid the `instanceof` check.

Reply:
Fair point, adjusted with [3d6b906](https://github.com/elastic/elasticsearch/pull/106476/commits/3d6b906e311e6929b5cfb65ef41300222b20f40f).